bool signal1
bool signal2
bool signal3
bool signal4
float32 dist
float32 compass
int32 state
